src/cpp/src/text_callback_streamer.cpp

```diff
@@ -15,6 +15,14 @@ bool TextCallbackStreamer::put(int64_t token) {
     std::stringstream res;
     m_tokens_cache.push_back(token);
     std::string text = m_tokenizer.decode(m_tokens_cache);
+    
+    // In some cases adding the next token can shorten the text, 
+    // e.g. when apostrophe removing regex had worked after adding new tokens.
+    // Need to hold on before flushing if apostrophe is the last symbol.
+    if (text.size() > 1 && text.back() == '\'') {
+        return on_finalized_subword_callback(res.str());
+    }
```

Comment:
There's one more implementation which should be fixed: https://github.com/openvinotoolkit/openvino.genai/blob/e2fa0d002117bbfe6d9a6a4cc413e583c4455131/samples/python/multinomial_causal_lm/multinomial_causal_lm.py#L11

Reply:
applied fix for it as well